e2e/tests/unit/utils/dataProvider.js

```diff
@@ -0,0 +1,91 @@
+  return Object.assign(load(`../../../../ccd-definition/${file}.json`), []);
+};
+
+let getFileData = [];
+let processDir = [];
+
+let fieldsArray = [];
+getFileData = (filePath, withConfig) => {
+  fieldsArray = [];
+  processDir(filePath, withConfig);
+  return fieldsArray;
+};
+
+processDir = (filePath, withConfig) => {
+  const fileNames = fs.readdirSync(path.resolve(__dirname, filePath));
+  if (!Object.prototype.toString.call(fileNames) === '[object Array]') {
+    const currentObject = path.resolve(__dirname, `${filePath}/${fileNames}`);
+    const stat = fs.statSync(currentObject);
+    if (stat.isFile() && !fileNames.filter(name => ['-HNL-nonprod.json', 'CUI.json'].includes(name)).length > 0) {
+      // if prod, exclude nonprod files
+      if (withConfig === '-prod.json') {
+        if (!fileNames.includes('-nonprod.json')) {
+          const content = Object.assign(load(currentObject), []);
+          fieldsArray = [...fieldsArray, ...content];
+        }
+      } else {
+        if (!fileNames.includes('-prod.json')) {
+          const content = Object.assign(load(currentObject), []);
+          fieldsArray = [...fieldsArray, ...content];
+        }
+      }
+    } else if (stat.isDirectory()) {
+      processDir(currentObject, withConfig);
+    }
+  } else {
+    fileNames.forEach(filename => {
+      const currentObject = path.resolve(__dirname, `${filePath}/${filename}`);
+      const stat = fs.statSync(currentObject);
+      if (stat.isFile() && !(['CUI.json'].filter(ext => filename.includes(ext)).length > 0)) {
+        if (withConfig === '-prod.json') {
+          if (!filename.includes('-nonprod.json')) {
+            const content = Object.assign(load(currentObject), []);
+            if (Object.prototype.toString.call(content) === '[object Array]') {
+              fieldsArray = [...fieldsArray, ...content];
+            }
+          }
+        } else {
+          if (!filename.includes('-prod.json')) {
+            const content = Object.assign(load(currentObject), []);
+            if (Object.prototype.toString.call(content) === '[object Array]') {
+              fieldsArray = [...fieldsArray, ...content];
+            }
+          }
+        }
+      } else if (stat.isDirectory()) {
+        processDir(currentObject, withConfig);
+      }
+    });
+  }
+};
+
+module.exports = {
+  ccdData: {
+    Banner: loadFile('Banner'),
+    CaseRoles: loadFile('CaseRoles'),
+    CaseType: loadFile('CaseType'),
+    Jurisdiction: loadFile('Jurisdiction'),
+    SearchCasesResultFields: loadFile('SearchCasesResultFields'),
+    SearchInputFields: loadFile('SearchInputFields'),
+    SearchResultFields: loadFile('SearchResultFields'),
+    State: loadFile('State'),
+    UserProfile: loadFile('UserProfile'),
+    WorkBasketInputFields: loadFile('WorkBasketInputFields'),
+    WorkBasketResultFields: loadFile('WorkBasketResultFields')
+  },
+  caseFieldata: getFileData('../../../../ccd-definition/CaseField', '-prod.json'),
```

Comment:
We don't have files with the `-prod.json` suffix at the time though, correct?

Basically: I'm trying to understand whether some of these changes are intended to work together with the other plans we have around using clearer naming conventions.

Reply:
@filippo-possenti-version1 this was WIP at the time, it's now parameterized to run with all exclusion scenarios defined in build-release-definition.sh